test/e2e/providers/azure/azure.go

```diff
@@ -0,0 +1,194 @@
+		TypeMeta: meta.TypeMeta{
+			APIVersion: "azureproviderconfig.openshift.io/v1beta1",
+			Kind:       "AzureMachineProviderSpec",
+		},
+		UserDataSecret: &core.SecretReference{
+			Name:      clusterinfo.UserDataSecretName,
+			Namespace: clusterinfo.MachineAPINamespace,
+		},
+		CredentialsSecret: &core.SecretReference{
+			Name:      defaultCredentialsSecretName,
+			Namespace: clusterinfo.MachineAPINamespace,
+		},
+		Location: location,
+		Zone:     &zone,
+		VMSize:   vmSize,
+		Image: azureprovider.Image{
+			Publisher: defaultImagePublisher,
+			Offer:     defaultImageOffer,
+			SKU:       defaultImageSKU,
+			Version:   defaultImageVersion,
+		},
+		OSDisk: azureprovider.OSDisk{
+			OSType:     "Windows",
+			DiskSizeGB: defaultOSDiskSizeGB,
+			ManagedDisk: azureprovider.ManagedDisk{
+				StorageAccountType: defaultStorageAccountType,
+			},
+		},
+		PublicIP:             false,
+		Subnet:               fmt.Sprintf("%s-worker-subnet", clusterID),
+		ManagedIdentity:      fmt.Sprintf("%s-identity", clusterID),
+		Vnet:                 fmt.Sprintf("%s-vnet", clusterID),
+		ResourceGroup:        rg,
+		NetworkResourceGroup: netrg,
+	}, nil
+}
+
+// GenerateMachineSet generates the machineset object which is aws provider specific
+func (p *Provider) GenerateMachineSet(withWindowsLabel bool, replicas int32) (*mapi.MachineSet, error) {
+	clusterID, err := p.oc.GetInfrastructureID()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get cluster id: %v", err)
+	}
+	platformStatus, err := p.oc.GetCloudProvider()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get azure platform status: %v", err)
+	}
+
+	// Inspect master-0 to get Azure Location and Zone
+	machines, err := p.oc.MachineClient.Machines("openshift-machine-api").Get(context.TODO(), clusterID+"-master-0", meta.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get master-0 machine resource: %v", err)
+	}
+	masterProviderSpec := new(azureprovider.AzureMachineProviderSpec)
+	err = json.Unmarshal(machines.Spec.ProviderSpec.Value.Raw, masterProviderSpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal master-0 azure machine provider spec: %v", err)
+	}
+
+	// create new machine provider spec for deploying Windows node in the same Location and Zone as master-0
+	providerSpec, err := newAzureMachineProviderSpec(clusterID, platformStatus, masterProviderSpec.Location, *masterProviderSpec.Zone, p.vmSize)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new azure machine provider spec: %v", err)
+	}
+
+	rawProviderSpec, err := json.Marshal(providerSpec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal azure machine provider spec: %v", err)
+	}
+
+	matchLabels := map[string]string{
+		mapi.MachineClusterIDLabel: clusterID,
+	}
+
+	machineSetName := "e2e-wmco"
+	if withWindowsLabel {
+		matchLabels[clusterinfo.MachineOSIDLabel] = "Windows"
+		// Designate machineSets with a windows label with "e2e-wmcow" in its name
```

Comment:
What does `wmcow` stand for? `Windows Machine Copy On Write`?

Reply:
I'm calling it `wmcow` to designate it as having the windows label (as opposed to just `wmco`). 
It's far from ideal, but I had to shorten the machine set name as otherwise the generated machine names were throwing an error for being too long. I'm still not clear why that doesn't seem to happen for Linux workers.

Reply:
I've clarified the naming scheme used here in the comment